genesyscloud/resource_genesyscloud_journey_segment_test.go

```diff
@@ -0,0 +1,103 @@
+package genesyscloud
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/mypurecloud/platform-client-sdk-go/v74/platformclientv2"
+)
+
+func TestAccResourceJourneySegmentSession(t *testing.T) {
+	runResourceJourneySegmentTestCase(t, "basic_session_attributes")
+}
+
+func TestAccResourceJourneySegmentCustomer(t *testing.T) {
```

Comment:
The update function isn't being called during these tests

Reply:
Why do you think that update isn't called during these tests? As I noted in other comment, in each test data folder we have multiple files where the first is the create step an than the following are the updates:

First test step is always the create step and all following steps are updates because the resource name is always the same. `generateTestSteps` generate test step array form these files for 1 test case, just as in already existing tests.
For example:
`test\data\resource\journey_segment\basic_customer_attributes\01_create_customer_scope.tf` has resource `"genesyscloud_journey_segment" "terraform_test_-TEST-CASE-" {`
This is the create step, than after that the update is:
`test\data\resource\journey_segment\basic_customer_attributes\02_external_segment_update_name.tf` with same resource name resource `"genesyscloud_journey_segment" "terraform_test_-TEST-CASE-" {`
and so on with `03_external_segment_force_new_creation.tf`

Here I have to note that meanwhile we developed this flow, we always debugged the `updateJourneySegment` and it was called at requested count depending on the update files count.

Reply:
![image](https://user-images.githubusercontent.com/107180950/176124230-f601fea0-297e-44ba-be05-28cdbcc18eff.png)
Just to show that update is executed.

Reply:
It must be something to do with how you're creating the config. It looks like a new resource is created instead of updating it.